hack/local-up-cluster.sh

```diff
@@ -688,111 +687,113 @@ function start_kubelet {
 
     priv_arg=""
     if [[ -n "${ALLOW_PRIVILEGED}" ]]; then
-      priv_arg="--allow-privileged=${ALLOW_PRIVILEGED} "
+      priv_arg="--allow-privileged=${ALLOW_PRIVILEGED}"
     fi
 
-    cloud_config_arg="--cloud-provider=${CLOUD_PROVIDER} --cloud-config=${CLOUD_CONFIG}"
+    cloud_config_arg=("--cloud-provider=${CLOUD_PROVIDER}" "--cloud-config=${CLOUD_CONFIG}")
     if [[ "${EXTERNAL_CLOUD_PROVIDER:-}" == "true" ]]; then
-       cloud_config_arg="--cloud-provider=external"
-       cloud_config_arg+=" --provider-id=$(hostname)"
+       cloud_config_arg=("--cloud-provider=external")
+       cloud_config_arg+=("--provider-id=$(hostname)")
     fi
 
     mkdir -p "/var/lib/kubelet" &>/dev/null || sudo mkdir -p "/var/lib/kubelet"
     # Enable dns
     if [[ "${ENABLE_CLUSTER_DNS}" = true ]]; then
       if [[ "${ENABLE_NODELOCAL_DNS:-}" == "true" ]]; then
-        dns_args="--cluster-dns=${LOCAL_DNS_IP} --cluster-domain=${DNS_DOMAIN}"
+        dns_args=("--cluster-dns=${LOCAL_DNS_IP}" "--cluster-domain=${DNS_DOMAIN}")
       else
-        dns_args="--cluster-dns=${DNS_SERVER_IP} --cluster-domain=${DNS_DOMAIN}"
+        dns_args=("--cluster-dns=${DNS_SERVER_IP}" "--cluster-domain=${DNS_DOMAIN}")
       fi
     else
       # To start a private DNS server set ENABLE_CLUSTER_DNS and
       # DNS_SERVER_IP/DOMAIN. This will at least provide a working
       # DNS server for real world hostnames.
-      dns_args="--cluster-dns=8.8.8.8"
+      dns_args=("--cluster-dns=8.8.8.8")
     fi
-    net_plugin_args=""
+    net_plugin_args=()
     if [[ -n "${NET_PLUGIN}" ]]; then
-      net_plugin_args="--network-plugin=${NET_PLUGIN}"
+      net_plugin_args=("--network-plugin=${NET_PLUGIN}")
     fi
 
-    auth_args=""
+    auth_args=()
```

Comment:
Was this a shellcheck recommended change? If not, would you mind elaborating on the benefits?

Reply:
auth_args have to be quoted for fixing the following shellcheck message.

In ./hack/local-up-cluster.sh line 773:
      ${auth_args}
      ^----------^ SC2206: Quote to prevent word splitting, or split robustly with mapfile or read -a.

If auth_args not defined as an array and dose not set any value, it was executed as blank option.

e.g.
hyperkube kubelet --allow-privileged=true ... --fail-swap-on=false '' --authorization-mode=Webhook